azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/ProxyConnectionHandler.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ */
+package com.microsoft.azure.eventhubs.impl;
+
+import com.microsoft.azure.proton.transport.proxy.ProxyHandler;
+import com.microsoft.azure.proton.transport.proxy.impl.ProxyHandlerImpl;
+import com.microsoft.azure.proton.transport.proxy.impl.ProxyImpl;
+
+import org.apache.qpid.proton.amqp.transport.ConnectionError;
+import org.apache.qpid.proton.amqp.transport.ErrorCondition;
+import org.apache.qpid.proton.engine.Connection;
+import org.apache.qpid.proton.engine.Event;
+import org.apache.qpid.proton.engine.Transport;
+import org.apache.qpid.proton.engine.impl.TransportInternal;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.Authenticator;
+import java.net.InetSocketAddress;
+import java.net.PasswordAuthentication;
+import java.net.Proxy;
+import java.net.ProxySelector;
+import java.net.URI;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ProxyConnectionHandler extends WebSocketConnectionHandler {
+    private static final Logger TRACE_LOGGER = LoggerFactory.getLogger(ProxyConnectionHandler.class);
+    private final String proxySelectorModifiedError = "ProxySelector has been modified.";
+
+    public static Boolean shouldUseProxy(final String hostName) {
+        final URI uri = createURIFromHostNamePort(hostName, ClientConstants.HTTPS_PORT);
+        final ProxySelector proxySelector = ProxySelector.getDefault();
+        if (proxySelector == null) {
+            return false;
+        }
+
+        final List<Proxy> proxies = proxySelector.select(uri);
+        return isProxyAddressLegal(proxies);
+    }
+
+    public ProxyConnectionHandler(AmqpConnection messagingFactory) {
+        super(messagingFactory);
+    }
+
+    @Override
+    protected void addTransportLayers(final Event event, final TransportInternal transport) {
+        super.addTransportLayers(event, transport);
+
+        final ProxyImpl proxy = new ProxyImpl();
+
+        // host name used to create proxy connect request
+        // after creating the socket to proxy
+        final String hostName = event.getConnection().getHostname();
+        final ProxyHandler proxyHandler = new ProxyHandlerImpl();
+        final Map<String, String> proxyHeader = getAuthorizationHeader();
+        proxy.configure(hostName, proxyHeader, proxyHandler, transport);
```

Comment:
>proxy.configure(hostName, proxyHeader, proxyHandler, transport); [](start = 8, length = 64)

Can configure handle Null for proxy header?

Reply:
yes, some cits does that.